sql/moz-fx-data-shared-prod/subscription_platform_derived/subplat_attribution_impressions_v1/query.sql

```diff
@@ -0,0 +1,156 @@
+    {% endif %}
+),
+service_flow_events AS (
+  {% if is_init() %}
+    SELECT
+      *
+    FROM
+      new_service_flow_events
+  {% else %}
+    SELECT
+      *
+    FROM
+      new_service_flow_events
+    UNION ALL
+    SELECT
+      flow_id,
+      flow_started_at AS flow_min_event_time,
+      impression_at AS event_time,
+      event_type,
+      entrypoint,
+      entrypoint_experiment,
+      entrypoint_variation,
+      utm_campaign,
+      utm_content,
+      utm_medium,
+      utm_source,
+      utm_term,
+      mozilla_account_id_sha256,
+      oauth_client_id,
+      oauth_client_name,
+      product_id,
+      plan_id,
+      service_id
+    FROM
+      `moz-fx-data-shared-prod.subscription_platform_derived.subplat_attribution_impressions_v1`
+    -- Unnesting these arrays can cause fan-outs, but the way we reaggregate things this doesn't matter.
+    LEFT JOIN
+      UNNEST(mozilla_account_ids_sha256) AS mozilla_account_id_sha256
+    LEFT JOIN
+      UNNEST(oauth_client_ids) AS oauth_client_id
+    LEFT JOIN
+      UNNEST(oauth_client_names) AS oauth_client_name
+    LEFT JOIN
+      UNNEST(product_ids) AS product_id
+    LEFT JOIN
+      UNNEST(plan_ids) AS plan_id
+    LEFT JOIN
+      UNNEST(service_ids) AS service_id
+  {% endif %}
+)
+SELECT
+  flow_id,
+  MIN(flow_min_event_time) AS flow_started_at,
+  ARRAY_AGG(
+    IF(
+      entrypoint_experiment IS NOT NULL
+      OR entrypoint_variation IS NOT NULL
+      OR utm_campaign IS NOT NULL
+      OR utm_content IS NOT NULL
+      OR utm_medium IS NOT NULL
+      OR utm_source IS NOT NULL
+      OR utm_term IS NOT NULL,
+      STRUCT(
+        event_time AS impression_at,
+        event_type,
+        entrypoint,
+        entrypoint_experiment,
+        entrypoint_variation,
+        utm_campaign,
+        utm_content,
+        utm_medium,
+        utm_source,
+        utm_term
+      ),
+      NULL
+    ) IGNORE NULLS
+    ORDER BY
+      event_time
+    LIMIT
+      1
+  )[ORDINAL(1)].*,
+  ARRAY_AGG(
+    DISTINCT mozilla_account_id_sha256 IGNORE NULLS
+    ORDER BY
+      mozilla_account_id_sha256
+  ) AS mozilla_account_ids_sha256,
+  ARRAY_AGG(DISTINCT oauth_client_id IGNORE NULLS ORDER BY oauth_client_id) AS oauth_client_ids,
+  ARRAY_AGG(
+    DISTINCT oauth_client_name IGNORE NULLS
+    ORDER BY
+      oauth_client_name
+  ) AS oauth_client_names,
+  ARRAY_AGG(DISTINCT product_id IGNORE NULLS ORDER BY product_id) AS product_ids,
+  ARRAY_AGG(DISTINCT plan_id IGNORE NULLS ORDER BY plan_id) AS plan_ids,
+  ARRAY_AGG(DISTINCT service_id IGNORE NULLS ORDER BY service_id) AS service_ids
+FROM
+  service_flow_events AS events
+GROUP BY
+  flow_id
+HAVING
+  LOGICAL_OR(
+    events.entrypoint_experiment IS NOT NULL
+    OR events.entrypoint_variation IS NOT NULL
+    OR events.utm_campaign IS NOT NULL
+    OR events.utm_content IS NOT NULL
+    OR events.utm_medium IS NOT NULL
+    OR events.utm_source IS NOT NULL
+    OR events.utm_term IS NOT NULL
+  )
```

Comment:
Unlike [`mozilla_vpn_derived.fxa_attribution_v1`](https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/mozilla_vpn_derived/fxa_attribution_v1), this will only keep flows with UTM/entrypoint attribution values, with the presumption being that we'll get attribution values before or at the same time as we get a `user_id`.

Reply:
will this cause a problem with impressions on date boundaries, including impact on `flow_min_event_time`?

Reply:
My thinking was if entrypoint/UTM parameters were going to be present in a flow they should be there from the start of the flow, but I suppose that isn't guaranteed.  I'll tweak this ETL to reprocess the previous day's data as well to try to catch all events in flows crossing date boundaries where the entrypoint/UTM parameters didn't appear until the subsequent date.